inc/Engine/Media/AboveTheFold/Frontend/Controller.php

```diff
@@ -74,11 +84,10 @@ public function lcp( $html ): string {
 		$row       = $this->query->get_row( $url, $is_mobile );
 
 		if ( empty( $row ) ) {
-			$this->manager->add_url_to_the_queue( $url, $is_mobile );
-			return $html;
+			return $this->inject_beacon( $html );
 		}
 
-		if ( 'completed' !== $row->status || empty( $row->lcp ) || 'not found' === $row->lcp ) {
+		if ( 'completed' !== $row->status || empty( $row->lcp ) || 'not found' === $row->lcp || 'failed' === $row->status ) {
```

Comment:
What if we added a new method in the row class here:
https://github.com/wp-media/wp-rocket/blob/b0d1b4e7ab95a6827e1f30270d856a5573a8763a/inc/Engine/Media/AboveTheFold/Database/Rows/AboveTheFold.php#L138

to be called `has_lcp` and has this logic, also we may use in_array to make conditions less than the current ones.

Reply:
That would be clever, I haven't thought of that. 
I've added a function like you said :)